libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -814,13 +830,13 @@ int lua_serial_readstring(lua_State *L) {
 
     // read up to that number of bytes
     const ssize_t read_bytes = port->read(data, req_bytes);
-    if (read_bytes < 0) {
-        return 0; // error, return nil
+    if (read_bytes >= 0) {
+        // push the buffer as a string, truncated to the number of bytes actually read
+        luaL_pushresultsize(&b, read_bytes);
+    } else {
+        lua_pushnil(L); // error, return nil
```

Comment:
You can just return 0 in this case. No need to push nil.

Reply:
Technically this does not actually result in returning nil. It's not a big deal here but is more of a problem for `port:read()`.